Products/ZenUI3/browser/eventconsole/export.py

```diff
@@ -75,11 +77,19 @@ def csv(self, response, archive, **params):
         response.setHeader('Content-Disposition', 'attachment; filename=events.csv')
         from csv import writer
         writer = writer(response)
-
+        events = [dict((k, v) for k, v in evt.iteritems() if v) for _, evt in
+                  self._query(archive, **params)]
+        keys = [x.keys() for x in events]
```

Comment:
This can be a generator of iterators, i.e.
```
keys = (x.iterkeys() for x in events)
```
and should still work with the next line.